src/road_cmd.cpp

```diff
@@ -1020,12 +1020,13 @@ CommandCost CmdBuildLongRoad(DoCommandFlags flags, TileIndex end_tile, TileIndex
 		}
 
 		CommandCost ret = Command<CMD_BUILD_ROAD>::Do(flags, tile, bits, rt, drd, TownID::Invalid());
-		if (ret.Failed() && ret.GetErrorMessage() == STR_ERROR_ALREADY_BUILT) ret = CommandCost(); // Treat "already built" as success
-
+		if (ret.GetErrorMessage() == STR_ERROR_ALREADY_BUILT) had_success = true;
 		if (ret.Failed()) {
 			last_error = std::move(ret);
-			if (is_ai) return last_error;
-			if (had_success) break; // Keep going if we haven't constructed any road yet, skipping the start of the drag
+			if (last_error.GetErrorMessage() != STR_ERROR_ALREADY_BUILT) {
```

Comment:
```suggestion
			if (last_error.GetErrorMessage() == STR_ERROR_ALREADY_BUILT) {
```

Reply:
The code you refer to is the existing code from before 13885 . This line is correct, your suggestion is wrong.